### PR TITLE
Bring 3.0 up-to-date with master.

### PR DESCRIPTION
**Pull request by Lukasa**
This was a gnarly merge, and the odds of it going right first time are pretty low, so let's have the CI prove or disprove the success of it for me.

### CODE REVIEW

**Comment by sigmavirus24** [comment]
@Lukasa you're merging this into master?

**Comment by nateprewitt** [comment]
Other than definitely not wanting to merge this into master, this looks like it's right. It would be nice to see a smaller diff against proposed/3.0.0 though.

**Comment by sigmavirus24** [comment]
Okay, I updated the branch for you

File: requests/__version__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# .-. .-. .-. . . .-. .-. .-. .-.
+# |(  |-  |.| | | |-  `-.  |  `-.
+# ' ' `-' `-`.`-' `-' `-'  '  `-'
+
+__title__ = 'requests'
+__description__ = 'Python HTTP for Humans.'
+__url__ = 'http://python-requests.org'
+__version__ = '2.17.3'
+__build__ = 0x021703
```
Comment (nateprewitt) [inline requests/__version__.py]: 0x030000

File: requests/__version__.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# .-. .-. .-. . . .-. .-. .-. .-.
+# |(  |-  |.| | | |-  `-.  |  `-.
+# ' ' `-' `-`.`-' `-' `-'  '  `-'
+
+__title__ = 'requests'
+__description__ = 'Python HTTP for Humans.'
+__url__ = 'http://python-requests.org'
+__version__ = '2.17.3'
```
Comment (nateprewitt) [inline requests/__version__.py]: This should to be 3.0.0, I believe.

File: requests/models.py [hunk 3]
```diff
@@ -335,17 +337,11 @@ def prepare_method(self, method):
         self.method = method
         if self.method is None:
             raise ValueError('Request method cannot be "None"')
-        self.method = to_native_string(self.method).upper()
+        self.method = to_native_string(self.method.upper())
```
Comment (nateprewitt) [inline requests/models.py:340]: Looks like parens may have been unintentionally moved here? I don't know if it's actually going to affect the final output but may cause issues in future merges.

Comment (Lukasa) [reply requests/models.py:340]: I *think* it doesn't matter? `.upper()` is safe on all types of strings.

Comment (nateprewitt) [reply requests/models.py:340]: Yep, this is purely stylistic I think. Just wanted to make sure we knew the change was being reverted.

File: requests/__version__.py (hunk 2 above)
Comment (sigmavirus24) [reply requests/__version__.py]: 👍

**Comment by Lukasa** [comment]
FYI, the windows tests won't finish because they're blocked. I pinged @kennethreitz about this in IRC.

**Comment by Lukasa** [comment]
Ok, the Windows CI won't get working any time soon as @kennethreitz is on the road. I suggest we move on without it to unblock #4118.

**Comment by nateprewitt** [comment]
@Lukasa, we have a [successful run](http://jenkins.kennethreitz.org:8080/job/requests/job/requests-windows/lastSuccessfulBuild/) of master on Windows from yesterday. Since nothing has changed on master, and I don't see anything directly related to Windows in the 3.0.0 merge, I think it's probably OK to merge this.

**Comment by Lukasa** [comment]
Kenneth restarted the builder, so we should get a result shortly.

**Comment by kennethreitz** [comment]
windows had a failure!

**Comment by Lukasa** [comment]
It did, that's true, but that failure seems to be related to having had something wacky happen with the environment build. I'm restarting the builder.

**Comment by Lukasa** [comment]
Yeah, I don't know that I trust the output of this Windows builder. It seems to not really be working properly.

File: .travis.yml [hunk 4]
```diff
@@ -16,7 +16,17 @@ install: "make"
 script:
   - |
     if [[ "$TRAVIS_PYTHON_VERSION" != "2.6" ]] ; then make test-readme; fi
-  - make coverage
-after_success:
-  - pipenv run codecov
+  - make ci
 cache: pip
+jobs:
```
Comment (WhyNotHugo) [inline .travis.yml:21]: You don't want the jobs AND the script above, it's kinda redundant.

Comment (WhyNotHugo) [reply .travis.yml:21]: Oh, this is actually the same on master though, so it's probably worth fixing there.

**Comment by Lukasa** [comment]
@kennethreitz What are we doing about this Windows builder?

**Comment by kennethreitz** [comment]
@Lukasa not sure why the build is hanging on Windows. I'll take a peek around. It runs fine on master.

**Comment by nateprewitt** [comment]
Looks like we're good to go here.

**Comment by WhyNotHugo** [comment]
I've rebased #4118 onto this locally, should I push that update?

**Comment by Lukasa** [comment]
@hobarrera Go for it

**Pull request merged**